### PR TITLE
LPS-173632 Create Workflow for Application Underwriting

### PR DESCRIPTION
**Pull request by joanagrego**
Related with --> [LPS-167987](https://issues.liferay.com/browse/LPS-167987)

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

**Review by nNilton — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+import java.util.Map;
+
+long classPK = GetterUtil.getLong((String)workflowContext.get(
+WorkflowConstants.CONTEXT_ENTRY_CLASS_PK));
+
+ObjectEntry objectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(classPK);
+
+Map<String, Serializable> values = objectEntry.getValues();
+
+Serializable applicationStatus = values.get("applicationStatus");
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:43]: You just use once, you can put this as inline

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+String status = applicationStatus.toString();
+
+if(status.equals("underwriting")){
+returnValue = "yes";
+}else {
+returnValue = "no";
+}]]>
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:51]: Format the code!

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+	values.put("applicationStatus", "reviewed");
+
+	long companyId = serviceContext.getCompanyId();
+
+	long userId = UserLocalServiceUtil.getUserIdByEmailAddress(companyId, "ryan.underwriter@mailinator.com");
+
+	ObjectEntryLocalServiceUtil.updateObjectEntry(userId,classPK,values, serviceContext);
+
+	WorkflowStatusManagerUtil.updateStatus(WorkflowConstants.getLabelStatus("approved"), workflowContext);]]>
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:153]: Format Code!

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+	ObjectEntry objectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(classPK);
+
+	Map<String, Serializable> values = objectEntry.getValues();
+
+	values.put("applicationStatus", "reviewed");
+
+	long companyId = serviceContext.getCompanyId();
+
+	long userId = UserLocalServiceUtil.getUserIdByEmailAddress(companyId, "ryan.underwriter@mailinator.com");
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:149]: You can use companyId as inline and the same for the userId in the next line

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+
+	values.put("applicationStatus", "rejected");
+
+	long companyId = serviceContext.getCompanyId();
+
+	long userId = UserLocalServiceUtil.getUserIdByEmailAddress(companyId, "ryan.underwriter@mailinator.com");
+
+	ObjectEntryLocalServiceUtil.updateObjectEntry(userId,classPK,values, serviceContext);
+
+WorkflowStatusManagerUtil.updateStatus(WorkflowConstants.getLabelStatus("denied"), workflowContext);]]>
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:210]: Format Code

**Review by nNilton — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+
+	ObjectEntry objectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(classPK);
+
+	Map<String, Serializable> values = objectEntry.getValues();
+
+	values.put("applicationStatus", "rejected");
+
+	long companyId = serviceContext.getCompanyId();
+
+	long userId = UserLocalServiceUtil.getUserIdByEmailAddress(companyId, "ryan.underwriter@mailinator.com");
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:206]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml (hunk 1 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:43]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml (hunk 4 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:149]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml (hunk 5 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:210]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml (hunk 3 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:153]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml (hunk 2 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/workflow-definitions/underwriting/workflow-definition.xml:51]: Done

**Comment by joanagrego** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
ci:forward

**Review by nNilton — APPROVED** [review]
LGTM

**Comment by joanagrego** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by joanagrego** [comment]
ci:forward

**Comment by EugenioAntunes** [comment]
Just starting reviewing :)

**Pull request closed without merging**